### PR TITLE
#211 BkTimetable wrong design

### PR DESCRIPTION
**Pull request by dmzaytsev**
For #211 

* BkTimetable has been refactored according the task description
* The unit test implemented as integration test
* The watcher's thread will be run until it become last in its thread group.

### CODE REVIEW

**Comment by davvd** [comment]
@dmzaytsev Thanks for the pull request, let me find a reviewer..

**Comment by davvd** [comment]
@darkled review this one, please

**Comment by darkled** [comment]
@rultor merge

**Comment by yegor256** [comment]
@dmzaytsev looks very good now, I'm sure you also see how easier is the test now and the class itself. I have only one question/concert - your method `isStop()`. I'm not entire against it, but don't you think that it's better to remove it and just call `setDaemon(true)` on the thread we're creating?

**Comment by dmzaytsev** [comment]
@yegor256 I agree the monitor thread should be a daemon.
But I created isStop() for other reason. It preventing it.file-manager integration test hung.
The monitor thread has to be stopped ever.

**Comment by dmzaytsev** [comment]
@yegor256 yes you're right  both the class and the test became easier.
But we use Thread.sleep() in the test. It is bad practice. @ggajos is not it? :)
I think we need a policy about using sleep() in unit tests. When permitted to use them and when not to.

**Comment by dmzaytsev** [comment]
@yegor256 I removed isStop()
take a look please

**Comment by yegor256** [comment]
@dmzaytsev sleeping in this particular test is perfectly legal

**Comment by yegor256** [comment]
@rultor merge

**Comment by dmzaytsev** [comment]
@rultor try to merge

@yegor256 I had no errors before push...
Maybe it is a temporary problem

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@dmzaytsev it's a legal failure, see https://ci.appveyor.com/project/yegor256/takes/build/827

**Comment by dmzaytsev** [comment]
@yegor256 I can't reproduce it locally..
Let me investigate

**Comment by dmzaytsev** [comment]
@yegor256 I will make some test  commits...

**Comment by dmzaytsev** [comment]
@yegor256 weird... I really don't understand correlation between FtBasicTest.gracefullyHandlesStuckBack() and BkTimeable
FtRemote doesn't use any BkTimeable calls
However AppVeyor build succeeded when a reverted BkTimeable.isStop()
Or this is really a temporary problem...

Let's try to merge ?

**Comment by dmzaytsev** [comment]
@yegor256 wait please
I will remove isStop() again

**Comment by dmzaytsev** [comment]
@yegor256 AppVeyor build succeeded with same conditions
FtBasicTest has a temporary problem and not reproducible.

**Comment by dmzaytsev** [comment]
@rultor try to merge

File: src/it/file-manager/pom.xml [hunk 1]
```diff
@@ -172,6 +172,7 @@
                                     <arguments>
                                         <argument>--port=${takes.port}</argument>
                                         <argument>--daemon=yes</argument>
+                                        <argument>--lifetime=3000</argument>
```
Comment (yegor256) [inline src/it/file-manager/pom.xml]: this shouldn't be here. test should work without this line

**Comment by yegor256** [comment]
@dmzaytsev I added one important comment above

**Comment by dmzaytsev** [comment]
@yegor256 fixed

**Comment by yegor256** [comment]
@rultor merge

**Comment by davvd** [comment]
@elenavolokhova please, let us know what do you think about this ticket, according to our [QA rules](http://www.teamed.io/qa.html)

**Comment by davvd** [comment]
@rultor deploy now pls

**Comment by elenavolokhova** [comment]
@darkled 
 There were no issues found during review. We should get used to this rule as soon as possible.
Please confirm that you will try to find at least three problems next time.

**Comment by darkled** [comment]
@elenavolokhova sure, I'll try to find at least three problems next time.

**Comment by elenavolokhova** [comment]
@davvd Quality is acceptable here.

**Comment by elenavolokhova** [comment]
@davvd Quality is acceptable.

**Comment by davvd** [comment]
> @davvd Quality is acceptable here.

@elenavolokhova thanks, got it. everybody, please try to make it good next time

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@elenavolokhova thanks for the review, we'll try better next time

**Comment by davvd** [comment]
> @davvd Quality is acceptable here.

@elenavolokhova thanks for the review, we'll try better next time

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@elenavolokhova thanks for the review, we'll try better next time

**Comment by davvd** [comment]
> @davvd Quality is acceptable here.

@elenavolokhova thanks for the QA review, we'll work better next time

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@elenavolokhova thanks a lot, next time everybody should try to make it better

**Comment by davvd** [comment]
> @davvd Quality is acceptable here.

@elenavolokhova thanks for the review, we'll try better next time

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@elenavolokhova thanks, got it. everybody, please try to make it good next time

**Comment by davvd** [comment]
> @davvd Quality is acceptable here.

@elenavolokhova thanks a lot, next time everybody should try to make it better

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@elenavolokhova thanks for the QA review, we'll work better next time

**Comment by davvd** [comment]
> @davvd Quality is acceptable here.

@elenavolokhova thanks, got it. everybody, please try to make it good next time

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@elenavolokhova thank you, let's try to make it "good" next time

**Comment by davvd** [comment]
> @davvd Quality is acceptable here.

@elenavolokhova thanks a lot, next time everybody should try to make it better

**Comment by davvd** [comment]
@darkled I added 10 mins to @elenavolokhova (for QA review) in transaction 56856857. Thanks a lot, I just topped your account for **17 mins**, transaction ID `56856867`, total time was 33 hours and 9 mins. extra minutes for review comments (c=2). +17 added to your rating, current score is: [+2052](http://www.netbout.com/b/35091?open=rating)

**Pull request merged**